pkg/virtual/syncer/coordination/helpers.go

```diff
@@ -0,0 +1,228 @@
+	oldSyncerViewAnnotations := make(map[string]string)
+	for name, value := range old.GetAnnotations() {
+		if strings.HasPrefix(name, v1alpha1.InternalSyncerViewAnnotationPrefix) {
+			if keepSyncTarget(strings.TrimPrefix(name, v1alpha1.InternalSyncerViewAnnotationPrefix)) {
+				oldSyncerViewAnnotations[name] = value
+			}
+		}
+	}
+
+	newSyncerViewAnnotations := make(map[string]string)
+	for name, value := range new.GetAnnotations() {
+		if strings.HasPrefix(name, v1alpha1.InternalSyncerViewAnnotationPrefix) {
+			if keepSyncTarget(strings.TrimPrefix(name, v1alpha1.InternalSyncerViewAnnotationPrefix)) {
+				newSyncerViewAnnotations[name] = value
+			}
+		}
+	}
+	return !reflect.DeepEqual(oldSyncerViewAnnotations, newSyncerViewAnnotations)
+}
+
+func SyncerViewsChanged(old, new metav1.Object) bool {
+	return FilteredSyncerViewsChanged(old, new, func(key string) bool {
+		return true
+	})
+}
+
+func SyncerViewChanged(old, new metav1.Object, syncTargetKey string) bool {
+	return FilteredSyncerViewsChanged(old, new, func(key string) bool {
+		return syncTargetKey == key
+	})
+}
+
+type Object interface {
+	metav1.Object
+	runtime.Object
+}
+
+func UpstreamViewChanged(old, new Object, equality func(old, new interface{}) bool) bool {
+	old = old.DeepCopyObject().(Object)
+	new = new.DeepCopyObject().(Object)
+
+	oldAnnotations := make(map[string]string)
+	for name, value := range old.GetAnnotations() {
+		if !strings.HasPrefix(name, v1alpha1.InternalSyncerViewAnnotationPrefix) {
+			oldAnnotations[name] = value
+		}
+	}
+	old.SetAnnotations(oldAnnotations)
+
+	newAnnotations := make(map[string]string)
+	for name, value := range new.GetAnnotations() {
+		if strings.HasPrefix(name, v1alpha1.InternalSyncerViewAnnotationPrefix) {
+			newAnnotations[name] = value
+		}
+	}
+	new.SetAnnotations(newAnnotations)
+
+	return !equality(old, new)
+}
+
+type SyncerViewManager[T Object] interface {
+	GetFilteredSyncerViews(ctx context.Context, gvr schema.GroupVersionResource, upstreamResource T, keepSyncTarget func(key string) bool) (map[string]T, error)
+	GetSyncerViews(ctx context.Context, gvr schema.GroupVersionResource, upstreamResource T) (map[string]T, error)
+	GetSyncerView(ctx context.Context, gvr schema.GroupVersionResource, upstreamResource T, syncTargetKey string) (T, error)
+}
+
+var _ SyncerViewManager[Object] = (*syncerViewManager[Object])(nil)
+
+type syncerViewManager[T Object] struct {
+	transformations.SyncerResourceTransformer
+}
+
+func NewSyncerViewManager[T Object](transformationProvider transformations.TransformationProvider,
+	summarizingRulesprovider transformations.SummarizingRulesProvider) SyncerViewManager[T] {
+	return &syncerViewManager[T]{
+		transformations.SyncerResourceTransformer{
+			TransformationProvider:   transformationProvider,
+			SummarizingRulesProvider: summarizingRulesprovider,
+		},
+	}
+}
+
+func NewDefaultSyncerViewManager[T Object]() SyncerViewManager[T] {
```

Comment:
what is a default sync manager? How is it default?

Reply:
Added documentation explains it